src/components/Snippet.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Accessor,
+  createContext,
+  createEffect,
+  createSignal,
+  PropsWithChildren,
+  Show,
+  useContext,
+} from "solid-js";
+import { isServer } from "solid-js/web";
+
+type Mode = "jsx" | "tsx" | undefined;
+
+const DisplayModeContext = createContext<Accessor<Mode>>(() => "jsx");
+
+const getDisplayMode: () => Mode = () => {
+  return !isServer && localStorage.getItem("mode") === "tsx" ? "tsx" : "jsx";
+};
+
+const Container = (props: PropsWithChildren) => {
```

Comment:
@nksaraf I think I'm having a hydration issue with this component that I'm using to toggle between JSX and TSX code snippets. Can you give me any advice? The `displayMode` defaults to `"jsx"` on the server. On the client, I grab the value from local storage. If local storage is in agreement with the default server state, no problem. But if localstorage is set to `"tsx"`, I get the following error: 

>Uncaught TypeError: Cannot read properties of undefined (reading 'cloneNode')